compilers/concrete-optimizer/concrete-optimizer/src/optimization/mod.rs

```diff
@@ -1,20 +1,26 @@
+use dag::multi_parameters::variance_constraint::VarianceConstraint;
+
 pub mod atomic_pattern;
 pub mod config;
 pub mod dag;
 pub mod decomposition;
 pub mod wop_atomic_pattern;
 
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, PartialEq)]
 pub enum Err {
     NotComposable(String),
     NoParametersFound,
+    UnfeasibleVarianceConstraint(VarianceConstraint),
 }
 
 impl std::fmt::Display for Err {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         match self {
             Self::NotComposable(details) => write!(f, "Program can not be composed: {details}"),
             Self::NoParametersFound => write!(f, "No crypto parameters could be found"),
+            Self::UnfeasibleVarianceConstraint(constraint) => {
+                write!(f, "Unfeasible noise constraint encountered: {constraint}")
```

Comment:
Ha ha curious to see the face of ML team with that. Maybe we can have a simplified print.